cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -417,6 +418,31 @@ private String getFileHash(final Path relativePath, final RevCommit commit)
     }
   }
 
+  private Map<Path, String> getFileHashes(Set<Path> relativePaths, RevCommit commit)
+      throws IOException, SourceControlException {
+    Map<Path, String> fileHashes = new HashMap<>();
+    if (relativePaths.size() == 0) {
+      return fileHashes;
+    }
+
+    try (TreeWalk walk = new TreeWalk(git.getRepository())) {
+      walk.addTree(commit.getTree());
+      walk.setRecursive(false);
+      while (walk.next()) {
+        if (walk.isSubtree()) {
+          walk.enterSubtree();
+        } else {
```

Comment:
nit: a little more readable to reduce one layer of nesting like:

```
if (walk.isSubtree()) {
  ...
  continue;
}
Path filePath...
```

Reply:
done